modules/git/git.go

```diff
@@ -161,68 +180,77 @@ func Init(ctx context.Context) error {
+	for configKey, defaultValue := range map[string]string{
+		"user.name":  "Gitea",
+		"user.email": "gitea@fake.local",
+	} {
+		if err := configSetNonExist(configKey, defaultValue); err != nil {
 			return err
 		}
 	}
 
 	// Set git some configurations - these must be set to these values for gitea to work correctly
-	if err := checkAndSetConfig("core.quotePath", "false", true); err != nil {
+	if err := configSet("core.quotePath", "false"); err != nil {
 		return err
 	}
 
 	if CheckGitVersionAtLeast("2.10") == nil {
-		if err := checkAndSetConfig("receive.advertisePushOptions", "true", true); err != nil {
+		if err := configSet("receive.advertisePushOptions", "true"); err != nil {
 			return err
 		}
 	}
 
 	if CheckGitVersionAtLeast("2.18") == nil {
-		if err := checkAndSetConfig("core.commitGraph", "true", true); err != nil {
+		if err := configSet("core.commitGraph", "true"); err != nil {
 			return err
 		}
-		if err := checkAndSetConfig("gc.writeCommitGraph", "true", true); err != nil {
+		if err := configSet("gc.writeCommitGraph", "true"); err != nil {
 			return err
 		}
 	}
 
 	if CheckGitVersionAtLeast("2.29") == nil {
 		// set support for AGit flow
-		if err := checkAndAddConfig("receive.procReceiveRefs", "refs/for"); err != nil {
+		if err := configAddNonExist("receive.procReceiveRefs", "refs/for"); err != nil {
 			return err
 		}
 		SupportProcReceive = true
 	} else {
-		if err := checkAndRemoveConfig("receive.procReceiveRefs", "refs/for"); err != nil {
+		if err := configUnsetAll("receive.procReceiveRefs", "refs/for"); err != nil {
 			return err
 		}
 		SupportProcReceive = false
 	}
 
-	if runtime.GOOS == "windows" {
-		if err := checkAndSetConfig("core.longpaths", "true", true); err != nil {
+	if CheckGitVersionAtLeast("2.35.2") == nil {
+		// since Git 2.35.2, git adds a protection for CVE-2022-24765, the protection denies the git directories which are not owned by current user
+		// however, some docker users and samba users (maybe more, issue #19455) have difficulty to set their Gitea git repositories to the correct owner.
+		// the reason behind the problem is: docker/samba uses some uid-mapping mechanism, which are unstable/unfixable in some cases.
+		// now Gitea always use its customized git config file, and all the accesses to the git repositories can be managed,
+		// so it's safe to set "safe.directory=*" for internal usage only.
+		if err := configSet("safe.directory", "*"); err != nil {
 			return err
 		}
 	}
-	if setting.Git.DisableCoreProtectNTFS {
-		if err := checkAndSetConfig("core.protectntfs", "false", true); err != nil {
+
+	if runtime.GOOS == "windows" {
+		if err := configSet("core.longpaths", "true"); err != nil {
 			return err
 		}
-		globalCommandArgs = append(globalCommandArgs, "-c", "core.protectntfs=false")
+		if setting.Git.DisableCoreProtectNTFS {
+			globalCommandArgs = append(globalCommandArgs, "-c", "core.protectntfs=false")
```

Comment:
We used to also set the config value here.

Reply:
There are no comment why we need both config set and command arguments.

Reply:
The code comes from #17300  , it seems not necessary repeat the config at both places.

If let me decide again, I would to choose to put the config in global config, then keep the arguments clear.

How do you think? (I will update the PR later)